### PR TITLE
handle BacklogLimitExceeded exception

### LINKED ISSUE #914: BacklogLimitExceeded on csv import
```python
Internal Server Error: /api/v1/forms/[pk]/csv_import.json

BacklogLimitExceeded at /api/v1/forms/[pk]/csv_import.json
ea3c17c4-373e-446f-9e29-e5af265f9641

Request Method: GET
Request URL: https://api.ona.io/api/v1/forms/[pk]/csv_import.json?job_uuid=ea3c17c4-373e-446f-9e29-e5af265f9641
Django Version: 1.9.5
Python Executable: /..../.virtualenvs/onadata/bin/uwsgi
Python Version: 2.7.6
Python Path: ['/..../.virtualenvs/onadata/src/savreaderwriter/savReaderWriter', '/..../.virtualenvs/onadata/src/savreaderwriter/savReaderWriter/cWriterow', '.', '', '/..../.virtualenvs/onadata/src/django-nose', '/..../.virtualenvs/onadata/src/django-digest', '/..../.virtualenvs/onadata/src/j2xclient', '/..../.virtualenvs/onadata/src/cors-headers', '/..../.virtualenvs/onadata/src/pricing', '/..../.virtualenvs/onadata/src/savreaderwriter', '/..../.virtualenvs/onadata/src/google-export', '/..../.virtualenvs/onadata/lib/python2.7', '/..../.virtualenvs/onadata/lib/python2.7/plat-x86_64-linux-gnu', '/..../.virtualenvs/onadata/lib/python2.7/lib-tk', '/..../.virtualenvs/onadata/lib/python2.7/lib-old', '/..../.virtualenvs/onadata/lib/python2.7/lib-dynload', '/usr/lib/python2.7', '/usr/lib/python2.7/plat-x86_64-linux-gnu', '/usr/lib/python2.7/lib-tk', '/..../.virtualenvs/onadata/local/lib/python2.7/site-packages']
Server time: Mon, 6 Feb 2017 09:55:35 -0500
Installed Applications:
('google_export',
 'pricing',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.sites',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.humanize',
 'django.contrib.admin',
 'django.contrib.admindocs',
 'django.contrib.gis',
 'registration',
 'django_nose',
 'django_digest',
 'corsheaders',
 'oauth2_provider',
 'rest_framework',
 'rest_framework.authtoken',
 'taggit',
 'readonly',
 'onadata.apps.logger',
 'onadata.apps.viewer',
 'onadata.apps.main',
 'onadata.apps.restservice',
 'onadata.apps.api',
 'guardian',
 'djcelery',
 'onadata.apps.sms_support',
 'onadata.libs',
 'reversion')
Installed Middleware:
('multidb.middleware.PinningRouterMiddleware',
 'onadata.libs.profiling.sql.SqlTimingMiddleware',
 'django.middleware.http.ConditionalGetMiddleware',
 'django.middleware.cache.UpdateCacheMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.cache.FetchFromCacheMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'onadata.libs.utils.middleware.LocaleMiddlewareWithTweaks',
 'django.middleware.csrf.CsrfViewMiddleware',
 'corsheaders.middleware.CorsMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'onadata.libs.utils.middleware.HTTPResponseNotAllowedMiddleware',
 'readonly.middleware.DatabaseReadOnlyMiddleware')


Traceback:

File "/..../.virtualenvs/onadata/local/lib/python2.7/site-packages/django/core/handlers/base.py" in get_response
  149.                     response = self.process_exception_by_middleware(e, request)

File "/..../.virtualenvs/onadata/local/lib/python2.7/site-packages/django/core/handlers/base.py" in get_response
  147.                     response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/..../.virtualenvs/onadata/local/lib/python2.7/site-packages/django/views/decorators/csrf.py" in wrapped_view
  58.         return view_func(*args, **kwargs)

File "/..../.virtualenvs/onadata/local/lib/python2.7/site-packages/rest_framework/viewsets.py" in view
  87.             return self.dispatch(request, *args, **kwargs)

File "/..../.virtualenvs/onadata/local/lib/python2.7/site-packages/rest_framework/views.py" in dispatch
  466.             response = self.handle_exception(exc)

File "/..../.virtualenvs/onadata/local/lib/python2.7/site-packages/rest_framework/views.py" in dispatch
  463.             response = handler(request, *args, **kwargs)

File "./onadata/apps/api/viewsets/xform_viewset.py" in csv_import
  549.                     request.query_params.get('job_uuid')))

File "./onadata/libs/utils/csv_import.py" in get_async_csv_submission_status
  303.     result = (job.result or job.state)

File "/..../.virtualenvs/onadata/local/lib/python2.7/site-packages/celery/result.py" in result
  356.         return self._get_task_meta()['result']

File "/..../.virtualenvs/onadata/local/lib/python2.7/site-packages/celery/result.py" in _get_task_meta
  339.             return self._maybe_set_cache(self.backend.get_task_meta(self.id))

File "/..../.virtualenvs/onadata/local/lib/python2.7/site-packages/celery/backends/amqp.py" in get_task_meta
  180.                 raise self.BacklogLimitExceeded(task_id)

Exception Type: BacklogLimitExceeded at /api/v1/forms/[pk]/csv_import.json
Exception Value: ea3c17c4-373e-446f-9e29-e5af265f9641
Request information:
GET:
job_uuid = u'ea3c17c4-373e-446f-9e29-e5af265f9641'

POST: No POST data

FILES: No FILES data

COOKIES: No cookie data
```

### PR DESCRIPTION
**Pull request by ukanga**
BacklogLimitExceeded exception eventually stops, hence returning a `PENDING` status allows continued checking since the export/import is usually still in progress, the task is not cancelled.

Fixes #914

### CODE REVIEW

File: onadata/libs/tests/utils/test_csv_import.py [hunk 1]
```diff
@@ -82,11 +87,13 @@ def test_submit_csv_and_rollback(self):
 
         count = Instance.objects.count()
         csv_import.submit_csv(self.user.username, self.xform, self.good_csv)
-        self.assertEqual(Instance.objects.count(),
-                         count + 9, u'submit_csv test Failed!')
+        self.assertEqual(Instance.objects.count(), count + 9,
```
Comment (pld) [inline onadata/libs/tests/utils/test_csv_import.py]: 9 is pretty magical number

File: onadata/libs/tests/utils/test_csv_import.py [hunk 2]
```diff
@@ -82,11 +87,13 @@ def test_submit_csv_and_rollback(self):
...
+                         u'submit_csv test Failed!')
         # Check that correct # of submissions belong to our user
-        self.assertEqual(Instance.objects.filter(user=self.user).count(),
-                         count + 8, u'submit_csv username check failed!')
+        self.assertEqual(
+            Instance.objects.filter(user=self.user).count(),
+            count + 8,
```
Comment (pld) [inline onadata/libs/tests/utils/test_csv_import.py]: but then again, so is 8

Comment (pld) [inline onadata/libs/tests/utils/test_csv_import.py]: but these are tests so I guess it's fine

Comment (ukanga) [inline onadata/libs/tests/utils/test_csv_import.py]: This has been around, was just applying https://github.com/google/yapf and `Isort` on files I touch.

Comment (ukanga) [inline onadata/libs/tests/utils/test_csv_import.py]: d1607c5cdaa376b8000933e588e534bf7b583ba8, f9bafa8a7fa6208b6a1d7c19891cbeb65ef65855, 14b544db3aec5dc470c5ba076d87178824cf36b9 and f47b864b83db956adff7f2fbb4f7770f375cd9fe are the key commits that are relevant in tackling #914.

**Comment by pld** [comment]
:+1:

**Pull request merged**